demo video link

aaronespinoza opened the pull request:
## What?
I added three elements, the left/right player paddle and the ball. I also rendered the respective score of each player. 
## Why?
These changes enable a player or players to interact with the respective paddles and run the game.
## How?
I rendered the ball and paddle elements to the browser using nodes and the player score via drawText function with canvas. On keydown events for both players, W and S for one player, Up and Down for another player the paddle moves along the y-axis by adding or subtracting the y value of the respective paddle. By pressing the SPACE button the game starts and pauses.  The ball bounces from boundaries by flipping the direction it was going if it's coordinates go past the height of the canvas.
## Testing?
An edge case that isn't resolved is when the ball hits the top of a players paddle.

aaronespinoza committed:
demo video link

demiculus commented:
@aaronespinoza well done with the functionality but there are problems with how it is presented (PRs) and how it is coded (cleanliness). I want you to do an iteration on top of it

1. You created your PR here https://github.com/colonistio/test/pull/31, then created another one here. You should have only created in your personal repository. It should have included the commits that I needed to review. Read upon it  https://medium.com/@hugooodias/the-anatomy-of-a-perfect-pull-request-567382bb6067
2. Break your functions into smaller pieces. Most functions should be 4-8 lines of code
3. Put stuff thats used multiple times inside variables. Also google magic number and don't use magic numbers.
4. Game is dull, add 1 interesting new feature on top of the refactors

Tag me in the new PR when its done. Good luck

aaronespinoza commented:
Good evening Demi,

I refactored the application and reduced the function sizes. I also removed
the magic numbers from the app. The two new features I added were a
difficulty toggle, allowing the player to increase or decrease the speed of
the ball and I also made the ball an emoji that changes on collisions.
Lastly I improved my PR and invited you to the repo for you to review. I'm
looking forward to see what you think of the new features Demi.

very respectfully,
Aaron Espinoza

On Fri, May 6, 2022 at 7:48 PM Demi Yilmaz ***@***.***> wrote:

> @aaronespinoza <https://github.com/aaronespinoza> well done with the
> functionality but there are problems with how it is presented (PRs) and how
> it is coded (cleanliness). I want you to do an iteration on top of it
>
>    1. You didn't create a decent PR. Read up on how a PR is created and
>    what it should include. I had to create this in order to review the code
>    changes: colonistio#32 <https://github.com/colonistio/test/pull/32>
>    ***@***.***/the-anatomy-of-a-perfect-pull-request-567382bb6067
>    2. Break your functions into smaller pieces. Most functions should be
>    4-8 lines of code
>    3. Put stuff thats used multiple times inside variables. Also google
>    magic number and don't use magic numbers.
>    4. Game is dull, add 1 interesting new feature on top of the refactors
>
> Tag me in the new PR when its done. Good luck
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/aaronespinoza/pingPong/pull/1#issuecomment-1120117202>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AWEHIIACHOUVKR3VATVK5EDVIXKW3ANCNFSM5VJOGYZA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>
